packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -457,36 +454,52 @@ export function TransferPanel() {
     }
   }, [amount, selectedToken, nativeCurrency])
 
+  const confirmUsdcDepositFromNormalOrCctpBridge = async () => {
+    const waitForInput = openUSDCDepositConfirmationDialog()
+    const [confirmed, primaryButtonClicked] = await waitForInput()
+
+    // user declined to transfer altogether
+    if (!confirmed) {
+      return false
+    }
+
+    // user has selected normal bridge (USDC.e)
+    if (primaryButtonClicked === 'bridged') {
+      return 'bridge-normal-usdce'
+    }
+
+    // user wants to bridge to native usdc using Circle's CCTP on destination chain
+    return 'bridge-cctp-usd'
+  }
+
+  const confirmUsdcWithdrawalForCctp = async () => {
+    const waitForInput = openUSDCWithdrawalConfirmationDialog()
+    const [confirmed] = await waitForInput()
+    return confirmed
+  }
+
+  const tokenAllowanceApprovalCctp = async () => {
+    setIsCctp(true)
+    const waitForInput = openTokenApprovalDialog()
+    const [confirmed] = await waitForInput()
+    return confirmed
+  }
+
   // SC wallet transfer requests are sent immediately, delay it to give user an impression of a tx sent
   const showDelayedSCTxRequest = () =>
     setTimeout(() => {
       setTransferring(false)
       setShowSCWalletTooltip(true)
     }, 3000)
 
-  const transferCctp = async (type: 'deposits' | 'withdrawals') => {
+  const transferCctp = async () => {
     if (!selectedToken) {
       return
     }
     if (!walletAddress) {
       return
     }
-
-    try {
-      setTransferring(true)
-      if (chainId !== networks.sourceChain.id) {
-        await switchNetworkAsync?.(networks.sourceChain.id)
-      }
-    } catch (e) {
-      if (isUserRejectedError(e)) {
-        return
-      }
-    } finally {
-      setTransferring(false)
-    }
-
-    const isDeposit = type === 'deposits'
-    const signer = isDeposit ? l1Signer : l2Signer
+    const signer = isDepositMode ? l1Signer : l2Signer
```

Comment:
We have this, but we also have:

```
  const { data: signer } = useSigner()
```

which we should probably get rid of?

Reply:
Yeah updated. It was not being used anywhere.